src/app/plan/duck/operations.ts

```diff
@@ -30,82 +30,170 @@ const pvFetchRequest = Creators.pvFetchRequest;
+              const hasErrorCondition = !!matchingStage.status.conditions.some(
+                c => c.type === 'Failed'
+              );
+              //if status is done, success
+              if (hasSucceededCondition) {
+                dispatch(stagingSuccess(createMigRes.data.spec.migPlanRef.name));
+                dispatch(commonOperations.alertSuccessTimeout('Staging Successful'));
+                return 'SUCCESS';
+              } else if (hasErrorCondition) {
+                dispatch(stagingFailure());
+                dispatch(commonOperations.alertErrorTimeout('Failed Staging'));
+                return 'FAILURE';
+              }
+            }
+          }
+        }
         return;
-      }
+      };
 
-      const nextProgress = plan.status.progress + 10;
-      dispatch(Creators.updatePlanProgress(plan.planName, nextProgress));
-    }, 1000);
+      const params = {
+        asyncFetch: fetchPlansGenerator,
+        delay: 500,
+        callback: callbackFunction,
+      };
+
+      dispatch(Creators.startStatusPolling(params));
+      dispatch(Creators.updatePlanMigrations(groupedPlan));
+    } catch (err) {
+      dispatch(commonOperations.alertErrorTimeout(err));
+      dispatch(stagingFailure(err));
+    }
   };
 };
 
 const runMigration = plan => {
   return async (dispatch, getState) => {
     try {
       dispatch(Creators.initMigration(plan.MigPlan.metadata.name));
+      dispatch(commonOperations.alertProgressTimeout('Migration Started'));
       const { migMeta } = getState();
       const client: IClusterClient = ClientFactory.hostCluster(getState());
 
       const migMigrationObj = createMigMigration(
         uuidv1(),
         plan.MigPlan.metadata.name,
-        migMeta.namespace
+        migMeta.namespace,
+        false
       );
       const migMigrationResource = new MigResource(MigResourceKind.MigMigration, migMeta.namespace);
 
       //created migration response object
       const createMigRes = await client.create(migMigrationResource, migMigrationObj);
 
-      const groupPlan: any = response => {
-        const fullPlan = {
-          MigPlan: plan.MigPlan,
-        };
-        if (response.data.items.length > 0) {
-          const sortMigrations = migrationList =>
-            migrationList.sort((left, right) => {
-              return moment
-                .utc(right.metadata.creationTimestamp)
-                .diff(moment.utc(left.metadata.creationTimestamp));
-            });
-
-          const matchingMigrations = response.data.items.filter(
-            i => i.kind === 'MigMigration' && i.spec.migPlanRef.name === plan.MigPlan.metadata.name
-          );
-
-          fullPlan['Migrations'] = sortMigrations(matchingMigrations);
-        } else {
-          fullPlan['Migrations'] = [];
+      const migrationListResponse = await client.list(migMigrationResource);
+      const groupedPlan = groupPlan(plan, migrationListResponse);
+
+      const callbackFunction = response => {
```

Comment:
Same comment here as above. I wonder if the logic in this is shared between the two callback functions at all and could be pulled into another fn?

Reply:
yeah definitely some room for refactoring here. I will see what I can do